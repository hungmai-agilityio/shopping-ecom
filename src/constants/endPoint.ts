export const END_POINT = {
  HOME: '/',
  ABOUT_US: '/about',
  CONTACT: '/contact',
  SIGN_IN: '/sign-in',
  SIGN_UP: '/sign-up',
  WISHLIST: '/wishlist',
  PRODUCTS: '/products',
  PRODUCT: '/product',
  CATEGORY: '/category',
  USERS: '/users',
  ACCOUNT: '/account',
  CART: '/cart',
  CHECKOUT: '/checkout'
};

export const QUERY = {
  CART: 'cart',
  USER: 'users',
  WISHLIST: 'wishlist',
  PRODUCTS: 'products',
}