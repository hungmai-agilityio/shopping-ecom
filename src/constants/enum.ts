export enum TYPE {
  PRIMARY = 'primary',
  SECOND = 'second',
  THIRD = 'third'
}

export enum INPUT_TYPE {
  FILE = 'file',
  TEXT = 'text',
  PASSWORD = 'password',
  EMAIL = 'email'
}

export enum SIZE {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large'
}

export enum CELL {
  TD = 'td',
  TH = 'th'
}

export enum KEY {
  USER = 'user'
}

export enum TIME_STATUS {
  STALE_TIME = 1000 * 60 * 60, // milliseconds * seconds * minutes = 1 hour
  COOKIE_TIME = 60 * 60 * 24 * 2 // seconds * minutes * hour * day
}
