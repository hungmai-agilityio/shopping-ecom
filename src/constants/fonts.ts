import { Poppins, Roboto } from 'next/font/google';

export const popping = Poppins({
  subsets: ['latin'],
  weight: ['400', '500', '700']
});

export const roboto = Roboto({
  subsets: ['latin'],
  weight: ['400', '500', '700']
});
