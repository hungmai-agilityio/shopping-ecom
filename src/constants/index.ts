export * from '@/constants/enum';
export * from '@/constants/styles';
export * from '@/constants/fonts';
