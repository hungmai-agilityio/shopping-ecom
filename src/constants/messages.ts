export const MESSAGE_VALID = {
  INPUT_REQUIRED: '{field} is required',
  EMAIL_ERROR: 'Invalid email format',
  EMAIL_EXIST: 'Email already exists',
  PASSWORD_ERROR: 'Password must be at least 6 characters',
  CONFIRM_ERROR: 'Passwords do not match',
  NEW_PASS_ERROR: 'Password cannot be the same as the old password',
  UPLOAD: 'Only file .png, .jpg, .jpeg, .svg allowed',
};

export const MESSAGE_API = {
  SIGN_UP_ERROR: 'An error occurred. Please try again',
  SIGN_UP_SUCCESS: 'You have successfully registered',
  SIGN_IN_ERROR: 'email or password is incorrect',
  ADD_PRODUCT_SUCCESS: 'Add product success',
  UPDATE_PROFILE_SUCCESS: 'Update profile success',
  UPDATE_PROFILE_ERROR: 'Update profile error',
  DELETE_ADDRESS_SUCCESS: 'Delete address success',
  DELETE_ADDRESS_ERROR: 'Delete address error',
  UPDATE_CART_ERROR: 'Update cart error',
  UPDATE_CART_SUCCESS: 'Update cart success',
  UPDATE_QUANTITY: 'Add quantity +1 to existing products'
};
