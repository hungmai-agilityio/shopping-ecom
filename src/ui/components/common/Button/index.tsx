import { ButtonHTMLAttributes, ReactNode } from 'react';
import clsx from 'clsx';

// Constants
import {
  btnSizeClasses,
  btnTypeClasses,
  popping,
  SIZE,
  TYPE
} from '@/constants';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: TYPE.PRIMARY | TYPE.SECOND | TYPE.THIRD;
  children?: ReactNode;
  size?: SIZE.SMALL | SIZE.MEDIUM | SIZE.LARGE;
  styles?: string;
}

const Button = ({
  children,
  variant = TYPE.PRIMARY,
  size = SIZE.MEDIUM,
  disabled,
  onClick,
  styles,
  ...props
}: ButtonProps) => {
  return (
    <button
      className={clsx(
        `rounded font-medium md:text-base text-xs ${popping.className}`,
        btnSizeClasses[size],
        btnTypeClasses[variant],
        styles,
        disabled && 'cursor-not-allowed opacity-50'
      )}
      disabled={disabled}
      onClick={onClick}
      {...props}
    >
      {children}
    </button>
  );
};

export default Button;
