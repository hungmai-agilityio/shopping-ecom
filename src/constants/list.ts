import { END_POINT } from "@/constants/endPoint";

export const navItems = [
  { name: 'Home', url: END_POINT.HOME },
  { name: 'Contact', url: END_POINT.CONTACT },
  { name: 'About', url: END_POINT.ABOUT_US }
];

export const listAbout = [
  { name: 'My Account', url: END_POINT.PROFILE },
  { name: 'Login / Register', url: END_POINT.SIGN_IN },
  { name: 'Cart', url: END_POINT.CART },
  { name: 'Wishlist', url: END_POINT.WISHLIST }
]

export const listQuick = [
  { name: 'Privacy Policy', url: '' },
  { name: 'Terms Of Use', url: '' },
  { name: 'FAQ', url: '' },
  { name: 'Contact', url: END_POINT.CONTACT }
]

export const slideHero = [
  {
    id: 'hero-1',
    image: '/hero_iphone.png',
    content: 'Up to 10% off Voucher',
    title: 'IPhone 14 Series',
    icon: '/icon-iphone.svg'
  },
  {
    id: 'hero-2',
    image: '/hero-canon.png',
    content: 'Capture stunning photos every time.',
    title: 'Canon camera'
  },
  {
    id: 'hero-3',
    image: '/hero-shoes.png',
    content: 'Durable, stylish, all-day comfort',
    title: 'Adidas Sneakers'
  },
  {
    id: 'hero-4',
    image: '/hero-cooler.png',
    content: 'Max Cooling Power - Liquid Cooling for Your PC!',
    title: 'Advanced Liquid Cooling System'
  },
  {
    id: 'hero-5',
    image: '/hero-ps5.png',
    content: 'Level Up with PS5 - Buy Now',
    title: 'PS5',
    icon: '/logo-sony.svg'
  }
];

export const slideStaff = [
  {
    id: 'staff-1',
    image: '/lead.png',
    name: 'Tom Cruise',
    desc: 'Founder & Chairman',
  },
  {
    id: 'staff-1',
    image: '/manager.png',
    name: 'Emma Watson',
    desc: 'Managing Director',
  },
  {
    id: 'staff-1',
    image: '/designer.png',
    name: 'Will Smith',
    desc: 'Product Designer',
  }
]