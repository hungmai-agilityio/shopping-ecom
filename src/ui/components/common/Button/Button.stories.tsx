import { Meta, StoryObj } from '@storybook/react';

// Components
import Button from '@/ui/components/common/Button';

// Constants
import { SIZE, TYPE } from '@/constants';

const meta: Meta<typeof Button> = {
  component: Button,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    children: {
      description: 'Name of the button or Content inside '
    },
    variant: {
      description: 'Set type optional for Button'
    },
    size: {
      description: 'Set size optional for Button'
    },
    onClick: {
      description: 'Handle the event when clicking on the button'
    },
    disabled: {
      description: 'Set disabled button'
    },
    styles: {
      description: 'Set styles for buttons'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    children: 'Submit',
    variant: TYPE.PRIMARY,
    onClick: () => alert('submit'),
    styles: 'w-[156px]'
  }
};

export const Secondary: Story = {
  args: {
    children: 'Submit',
    variant: TYPE.SECOND,
    onClick: () => alert('submit'),
    styles: 'w-[156px]'
  }
};

export const Third: Story = {
  args: {
    children: 'Forgot password',
    variant: TYPE.THIRD,
    size: SIZE.SMALL
  }
};

export const Large: Story = {
  parameters: {
    layout: 'left'
  },
  args: {
    children: 'Submit',
    variant: TYPE.PRIMARY,
    size: SIZE.LARGE,
    onClick: () => alert('submit')
  }
};

export const Disabled: Story = {
  args: {
    children: 'Send',
    disabled: true,
    styles: 'w-[156px]'
  }
};
