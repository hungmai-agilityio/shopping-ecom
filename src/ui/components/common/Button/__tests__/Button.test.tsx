import { fireEvent, render, screen } from '@testing-library/react';

// Constants
import { TYPE } from '@/constants';

// Components
import Button from '@/ui/components/common/Button';

describe('Button', () => {
  const onClickMock = jest.fn();

  test('renders button with default props', () => {
    render(<Button children="Test Button" variant={TYPE.PRIMARY} />);

    const buttonElement = screen.getByText('Test Button');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveClass(
      'h-14 rounded font-medium md:text-base text-xs max-w-btn-md bg-primary text-white'
    );
    expect(buttonElement).not.toBeDisabled();
  });

  test('renders button with variant prop', () => {
    render(<Button children="Test Button" variant={TYPE.SECOND} />);

    const buttonElement = screen.getByText('Test Button');
    expect(buttonElement).toHaveClass(
      'h-14 rounded font-medium md:text-base text-xs max-w-btn-md bg-white'
    );
  });

  test('button is disabled when disabled prop is true', () => {
    render(<Button children="Test Button" disabled />);

    const buttonElement = screen.getByText('Test Button');
    expect(buttonElement).toBeDisabled();
    expect(buttonElement).toHaveClass('cursor-not-allowed');
  });

  test('calls the onClick handler when clicked', () => {
    render(
      <Button variant={TYPE.SECOND} onClick={onClickMock}>
        Action
      </Button>
    );

    const buttonElement = screen.getByRole('button', { name: /action/i });
    fireEvent.click(buttonElement);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  test('should be render match to snapshot', () => {
    const { container } = render(
      <Button children="Click me" variant={TYPE.PRIMARY} />
    );

    expect(container).toMatchSnapshot();
  });
});
