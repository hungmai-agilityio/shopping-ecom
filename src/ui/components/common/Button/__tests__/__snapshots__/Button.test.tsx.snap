// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be render match to snapshot 1`] = `
<div>
  <button
    class="h-14 rounded font-medium md:text-base text-xs className max-w-btn-md bg-primary text-white"
  >
    Click me
  </button>
</div>
`;
